OpenRA.Game/TranslationProvider.cs

```diff
@@ -0,0 +1,46 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.FileSystem;
+
+namespace OpenRA
+{
+	public class TranslationProvider
+	{
+		static readonly List<Translation> TranslationProviders = new List<Translation>();
+
+		public static void Initialize(ModData modData, IReadOnlyFileSystem fileSystem)
+		{
+			Deinitialize();
+
+			TranslationProviders.Add(new Translation(Game.Settings.Player.Language, modData.Manifest.Translations, modData.DefaultFileSystem));
+			if (fileSystem is Map map)
+				TranslationProviders.Add(new Translation(Game.Settings.Player.Language, map.Translations, fileSystem));
+		}
+
+		public static void Deinitialize()
+		{
+			TranslationProviders.Clear();
+		}
+
+		public static string GetString(string key, IDictionary<string, object> args = null)
+		{
+			foreach (var translationProvider in TranslationProviders)
+			{
+				if (translationProvider.TryGetString(key, out var message, args))
+					return message;
+			}
```

Comment:
Can we change this from a list to two fields which are checked in succession? At the moment it is not clear that checking the map translations last is intentional and not a bug by just reading the code.

Reply:
A comment would fix that. Eg.:

// By prioritising mod level translation keys we make sure that they aren't overwritten by map level keys.

Reply:
I thought about that too, but relying on element order in a list is a bit volatile when we can easily be explicit without any downsides.

Reply:
I agree with the initial request. I could understand using a list if we had 3 or more sources, but with only two it is conceptually simpler to just have two `TryGetStrings`.